### PR TITLE
feat: allow runtieme configuration

### PR DESCRIPTION
**Pull request by dcoa**
**Description**

This PR is part of the work to make it possible to configure the frontend applications at runtime (you can referer to this https://github.com/openedx/frontend-wg/issues/103).

**Changes**

 - Update frontend-platform to version 2.5.0
 - Update the footer and header to avoid peer dependency errors.
 - Create a component Head, that uses Helmet library and integrates internationalization to change the MFE name in the title tag according to the language, and change the favicon in runtime.

Screenshot 
gradebook-title

Values come from runtime configuration 
response runtime-config

**How to test**

- To allow runtime configuration set a new environment variables `MFE_CONFIG_API_URL` and `APP_ID` in the env file and add the api url ( To test this you can use the API from this https://github.com/openedx/edx-platform/pull/30473 or use an external tool to mock the API response).
- The API should respond with a JSON with the config values, something like:
```json
{
  "SITE_NAME": "Test Site",
  "LOGO_URL": "https://testimage.com/logo.svg",
  "LOGO_TRADEMARK_URL": "https://testimage.com/logo.svg",
  "LOGO_WHITE_URL": "https://testimage.com/logo.svg",
  "FAVICON_URL": "https://testimage.com/favicon.ico",
}
```
- The initialize process should work normally.
_Note: You can combine buildtime and runtime configuration_

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @dcoa!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@dcoa Thank you for your contribution. Is this ready for our review?

**Comment by dcoa** [comment]
Hi @natabene , yes is ready.

**Comment by schenedx** [comment]
@dcoa I had a review on this PR. I would like to request the following:

- Could you share the business reason behind this change? Is this purely translation related? 
- Could you pictures on this PR? A picture describing the area of change on the UI, and what the effect of it is?
- Am I understanding correctly this repo is forked within your work environment, and you are just trying merge back the changes from your fork to help the main repo?

File: .env
Comment (schenedx) [inline .env:33]: Why are these config value needed? I don't see them used in this PR

File: src/App.jsx
Comment (schenedx) [inline src/App.jsx:32]: It's weird to me that `Head` element is lower than the `Footer` element. Can you help me understand the reason for this sequence?

File: src/head/Head.test.jsx
Comment (schenedx) [inline src/head/Head.test.jsx:10]: #NIT Please correct spelling on `favicon`

**Review by schenedx — DISMISSED** [review]
Added some inline comments as well.

**Comment by muselesscreator** [comment]
Please elaborate on the need for inclusion of a new react libary (react-helmet)?

File: src/App.jsx
Comment (dcoa) [reply src/App.jsx]: I agree, I fix this

**Comment by dcoa** [comment]
Hi @schenedx, 

> Could you share the business reason behind this change? Is this purely translation related?

The main idea form this PR is to allow gradebook to use the runtime configuration option when this is activated (now in frontend-platform check this PR https://github.com/openedx/frontend-platform/pull/335).

> Could you pictures on this PR? A picture describing the area of change on the UI, and what the effect of it is?

Yes, I added a screenshot of the area impacted by Head component the main idea is to maintain the actual UI.

The main idea of runtime configuration is to set or override any variable this includes  `SITE_NAME`  and `FAVICON_URL`, currenly, these variables are only read at build time https://github.com/openedx/frontend-app-gradebook/blob/master/public/index.html#L4 if I want to use runtime configuration and update the values those changes aren't applied.

@muselesscreator, understanding that, is necessary to apply these changes with code (accessing the method `getConfig`). After evaluate the different ways to do that, I chose react-helmet that has been tested and used in other MFEs like [frontend-app-learning](https://github.com/openedx/frontend-app-learning/blob/master/package.json#L54) and [fronted-app-auth](https://github.com/openedx/frontend-app-authn/blob/master/package.json#L58) to make changes in the title tag at runtime. 

> Am I understanding correctly this repo is forked within your work environment, and you are just trying to merge back the changes from your fork to help the main repo?

Yes, I tried to make a contribution to the main code. 


Like the issue https://github.com/openedx/frontend-wg/issues/103 describes I want to implement those changes in different apps (availables by default in tutor-mfe by now), that includes Gradebook and Account where was approved https://github.com/openedx/frontend-app-account/pull/603


I will be attentive if you require more information 🤓

File: src/App.jsx
Comment (dcoa) [reply src/App.jsx]: I use a component because there are specific related changes (title and favicon) and I thought was better to maintain a simple App component and because I'm using `injectIntl` (from `@edx/frontend-platform/i18n`) which wraps the component to make it possible to change the MFE name according to the language and that needs access the `IntlProvider` thai is inside `AppProvider`

**Comment by kdmccormick** [comment]
@schenedx and @muselesscreator -- I plan to approve and merge this on Monday 10/17 unless you folks had any specific concerns.

**Review by muselesscreator — APPROVED** [review]

**Comment by MaferMazu** [comment]
I tested it, and it works. Thanks for your explanation and the images @dcoa 🙌

**Review by kdmccormick — COMMENTED** [review]
@dcoa would you mind rebasing this one last time before we merge it?

@schenedx , did you want to re-review this at all?

File: .env.development
Comment (schenedx) [reply .env.development:41]: @kdmccormick No. If @muselesscreator approved. I am good.
@muselesscreator When is a good time for us to merge and deploy this change?

**Comment by dcoa** [comment]
Hi @kdmccormick I already rebase it and add the single quotes.

**Review by kdmccormick — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@dcoa 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-gradebook&entry.867055334=https://github.com/openedx/frontend-app-gradebook/pull/253&entry.1484655318=https://github.com/dcoa&entry.752974735=2022-07-13+19:32&entry.1917517419=2022-10-17+16:12&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
